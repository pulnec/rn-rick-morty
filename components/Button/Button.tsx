import { View, Text, ButtonProps, ActivityIndicator } from "react-native";
import React from "react";
import styled, { css } from "@emotion/native";

const ButtonRN = styled.Pressable`
  align-items: center;
  background-color: #000;
  align-self: center;
  padding: 8px 15px;
  border-radius: 8px;
`;

interface IButtonProps extends ButtonProps {
  loading?: boolean;
}

export default function Button(props: IButtonProps) {
  return (
    <ButtonRN {...props}>
      {props.loading ? (
        <ActivityIndicator testID="loading-button" size="small" color="#fff" />
      ) : (
        <Text testID="title-button" style={{ color: "#fff" }}>{props.title}</Text>
      )}
    </ButtonRN>
  );
}
