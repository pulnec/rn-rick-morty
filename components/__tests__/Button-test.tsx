import * as React from 'react';
import Button from '../Button/Button';
import { render } from '@testing-library/react-native';

it(`renders correctly`, () => {
    const tree = render(<Button title="Test" />).toJSON();
    expect(tree).toMatchSnapshot();
});

it(`check loading is visible`, () => {
    const { getByTestId } = render(<Button title="Test" loading={true} />);
    expect(getByTestId('loading-button')).toBeTruthy();
});

it(`check title prop`, () => {
    const { getByTestId } = render(<Button title="Test" />);
    expect(getByTestId('title-button').props.children).toBe('Test');
});