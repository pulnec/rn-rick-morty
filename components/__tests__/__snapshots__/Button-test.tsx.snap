// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#000",
        "borderBottomLeftRadius": 8,
        "borderBottomRightRadius": 8,
        "borderTopLeftRadius": 8,
        "borderTopRightRadius": 8,
        "paddingBottom": 8,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 8,
      },
      undefined,
    ]
  }
  title="Test"
>
  <Text
    style={
      {
        "color": "#fff",
      }
    }
    testID="title-button"
  >
    Test
  </Text>
</View>
`;
